src/lib/CustomRulePathManager.ts

```diff
@@ -15,30 +17,41 @@ export const CUSTOM_CLASSPATH_REGISTER_TMP = path.join(SFDX_SCANNER_PATH, `Tmp${
 
 const EMPTY_JSON_FILE = '{}';
 
-export class CustomRulePathManager {
+
+export class CustomRulePathManager extends AsyncCreatable {
+  private logger!: Logger;
   private pathsByLanguageByEngine: RulePathMap;
   private initialized: boolean;
   private fileHandler: FileHandler;
 
-  constructor() {
+
+  protected async init(): Promise<void> {
+    this.logger = await Logger.child('CustomRulePathManager');
     this.pathsByLanguageByEngine = new Map();
     this.fileHandler = new FileHandler();
     this.initialized = false;
   }
 
   private async initialize(): Promise<void> {
     if (this.initialized) {
+      this.logger.trace(`CustomRulePathManager has already been initialized`);
       return;
     }
 
+    this.logger.trace(`Initializing CustomRulePathManager.`);
+
     // Read from the JSON and use it to populate the map.
     let data = null;
     try {
-      data = await this.fileHandler.readFile(CustomRulePathManager.getFilePath());
+      const customRulePathFile = CustomRulePathManager.getFilePath();
+      this.logger.trace(`Reading CustomRulePath file at ${customRulePathFile}`);
+      data = await this.fileHandler.readFile(customRulePathFile);
+      this.logger.trace(`CustomRulePath content read as: ${data}`);
```

Comment:
I'm not sure why, but it strikes me as excessive for these to be two separate log statements instead of just one that says "Read the following data from this file: ...".

Reply:
Hmm, I suppose this is coming from a past experience where we weren't sure if an external operation was called or not while troubleshooting where the failure came from and we ended up adding logs like these. Though I think we should be fine adding more logs when we do need to troubleshoot. I can remove the log at line 47.